test/integration/shell_int_test.go

```diff
@@ -288,13 +291,19 @@ func (suite *ShellIntegrationTestSuite) TestNestedShellNotification() {
 
 	cfg, err := config.New()
 	suite.Require().NoError(err)
+	if runtime.GOOS == "darwin" && condition.OnCI() {
+		cfg.Set(subshell.ConfigKeyShell, "zsh") // GitHub actions runs on bash, so override with zsh
+	}
 
+	os.Setenv(constants.HomeEnvVarName, ts.Dirs.HomeDir)
+	defer func() { os.Unsetenv(constants.HomeEnvVarName) }()
```

Comment:
This is not safe for other tests running in parallel. You can set it with AppendEnv though.

Reply:
I've tried to minimize time spent in a modified env state. It's needed at the function-level for `ss.RcFile()` and spawn-process-level for `user.HomeDir()`.

Reply:
Ok, please find an alternative to using `ss.RcFile()` that doesn't require `os.Setenv()`. This is just a test so it's ok if we're hard coding some stuff here so long as it's not prone to breaking all the time.

As for the process-level; `AppendEnv()` should be able to handle that.

Minimizing the window doesn't change that it can impact other tests in ways that will be really hard to identify after the fact.

Reply:
I've hardcoded creating and populating shellrc files like the installer would.